crossbuilder

```diff
@@ -845,10 +958,79 @@ if ! which lxd > /dev/null ; then
     fi
 fi
 
-check_lxd_accessible
 ensure_lxd_subuid
+check_lxd_accessible
+
+if [ -z $HOST_ARCH ] ; then
+    if which dpkg > /dev/null 2>&1 ; then
+        HOST_ARCH=`dpkg --print-architecture`
+    else
+        # To find out the host's debian architecture name, first find out debian multiarch tuple and use conversion table.
+        # Source: https://wiki.debian.org/Multiarch/Tuples
+
+        if [ -z $HOST_FARCH ] && which gcc > /dev/null 2>&1 ; then
+            HOST_FARCH=$(gcc -print-multiarch)
+
+            # On manjaro the 'gcc -print-multiarch' returns nothing.
+            # Use the 'gcc -dumpmachine' multiarch triplet result and strip the vendor part if the result contains it.
+            [ -z $HOST_FARCH ] && HOST_FARCH=$(gcc -dumpmachine | sed 's/^\([^-]\+\)-\([^-]\+\)-\([^-]\+\)-\([^-]\+\)$/\1-\3-\4/')
+        fi
+        #echo "HOST_FARCH=\"${HOST_FARCH}\""
+
+        declare -A MULTIARCH_TO_DEBIAN=( \
```

Comment:
do you think it makes sense to use a similar trick for converting the missing package names?

Reply:
Where exactly do you mean? I had a feeling, when browsing code before, that this could be used somewhere, but now I cant remember or find where. :(

Reply:
Yes, I also did not remember clearly, because I thought that there were many places were we could have benefitted from a package map, but in fact the only two places are these:

1. https://github.com/jezek/crossbuilder/blob/manjaro/crossbuilder#L864-L868
2. https://github.com/jezek/crossbuilder/blob/manjaro/crossbuilder#L276-L281

I think that eventually we will want to have a function `install_host_package()` which takes a list of Ubuntu package names as input parameters, and converts them to other distro packages and installs them as needed, but we can leave it for a future change, as the current one is already big enough. :-)